onnxruntime/core/framework/debug_node_inputs_outputs_utils.cc

```diff
@@ -188,7 +189,11 @@ const NodeDumpOptions& NodeDumpOptionsFromEnvironmentVariables() {
 
     NodeDumpOptions opts{};
 
-    opts.dump_flags = NodeDumpOptions::DumpFlags::ShapeOnly;
+    // Preserve existing behavior of printing the shapes by default. Turn it off only if the user has requested so
+    // explicitly by setting the value of the env variable to 0.
+    const int default_dump_flags = NodeDumpOptions::DumpFlags::Shape;
+    opts.dump_flags = ParseEnvironmentVariable<int>(env_vars::kDumpShapeData, default_dump_flags);
```

Comment:
how about this:
```
opts.dump_flags = NodeDumpOptions::DumpFlags::None;
if (ParseEnvironmentVariable<bool>(env_vars::kDumpShapeData, true)) {
  opts.dump_flags |= NodeDumpOptions::DumpFlags::Shape;
}
```
If we set opts.dump_flags directly from the environment variable, it allows the user to set other flags too (e.g., by passing 0xF).